### PR TITLE
cilium-dbg: Expose Cilium network routing status

### PR DESCRIPTION
**Pull request by joestringer**
Expose the routing mode currently active in the cilium-dbg status CLI.

Example:
```
$ cilium-dbg status
...
Routing:                              Network: Tunnel [vxlan]   Host: Legacy
...
```

### CODE REVIEW

**Review by learnitall — APPROVED** [review]
Thanks Joe! I'm surprised we don't have this already.

The changes to the API are breaking, but since this API resource is only consumed by utilities which are built alongside the API, we shouldn't have to worry about breaking compatibility. It's not expected for an external consumer to rely on the existing of the `HostRouting` object, as external consumers of Cilium's API aren't expected in general, and we don't package different Cilium client and server versions into the same image.

**Comment by brb** [comment]
> `Host: Legacy`

A bit of bikeshedding, but ... IMO the BPF host routing is still an experimental feature (e.g., in the ENI mode it's discouraged). For me "legacy" reads as if something is bad / not wanted. Maybe we could improve the wording by just saying `BPF routing: (on | off)`.

**Comment by joestringer** [comment]
@brb I think that's fair criticism. Note we've been using this terminology for several releases now and I think it's actually been pretty effective at nudging users to try to upgrade their kernels and get BPF host routing enabled. That said I didn't have a strong opinion on this. @borkmann would you like to chime in on the above comment from Martynas?

**Review by thorn3r — APPROVED** [review]
Nice addition :+1: Agreed that `off` sounds a bit nicer than `legacy` if there are some cases where it will never be supported. If that wording has inspired FOMO and encouraged some folks to upgrade, I'm guessing it will also inspire the same for EKS users

**Comment by joestringer** [comment]
@brb I opened https://github.com/cilium/cilium/issues/32131 to follow up on that.

File: pkg/client/client.go [hunk 1]
```diff
@@ -491,8 +491,14 @@ func FormatStatusResponse(w io.Writer, sr *models.StatusResponse, sd StatusDetai
...
 
-	if sr.HostRouting != nil {
-		fmt.Fprintf(w, "Host Routing:\t%s\n", sr.HostRouting.Mode)
+	if sr.Routing != nil {
+		status := "Network: " + sr.Routing.InterHostRoutingMode
+		if sr.Routing.InterHostRoutingMode == models.RoutingInterHostRoutingModeTunnel {
+			status = status + " [" + sr.Routing.TunnelProtocol + "]"
+		}
+		status = status + "\tHost: " + sr.Routing.IntraHostRoutingMode
+
```
Comment (julianwiedmann) [inline pkg/client/client.go:500]: Thanks, that's been on my list for quite a while :slightly_smiling_face:. Did you consider also exposing the tunnel port?

**Pull request merged**